pkg/internal/diskutil.go

```diff
@@ -324,11 +324,12 @@ func GetDeviceFSMap() (map[string]string, error) {
 func GetPVCreationLock(device string, symlinkDirs ...string) (ExclusiveFileLock, bool, []string, error) {
 	lock := ExclusiveFileLock{Path: device}
 	locked, err := lock.Lock()
-	if err != nil || !locked {
+	if err != nil {
 		return lock, false, []string{}, err
 	}
 	existingLinkPaths, err := GetMatchingSymlinksInDirs(device, symlinkDirs...)
-	if err != nil || len(existingLinkPaths) > 0 {
+	// If we don't have any existing symlinks AND we can't get a lock, throw an error
+	if err != nil || (len(existingLinkPaths) == 0 && !locked) {
```

Comment:
But what is the error here then? if we couldn't get lock because device was busy and no existing symlinks were present then `err` variable will be nil.

Reply:
I've updated lock acquisition to return the error regardless of if the device is busy. This keeps the same logic, but populates the error if the device is busy (which isn't reported as long as there's one or more links).

Let me know your thoughts on this approach.